### ISSUE TITLE
yacc syntax breaks if comment on section seperator

### ISSUE THREAD

**Issue by mjsir911** [vim/vim#4164]
File looks like this:
```yacc
%{
#include <stdio.h>
%}

%token x /* why am I needed for syntax highlighting? */

%% /* syntax breaks because of me */
line        : /* empty */
            ;
%%

void yyerror(const char *str)
{
  fprintf(stderr,"error: %s\n",str);
}

int yywrap()
{
  return 1;
}

int main()
{
  printf("> "); 
  yyparse();
  return 0;
}
```

syntax highlighting breaks after line 7 due to comment `/* syntax breaks because of me */`

I believe this is caused by the line (and similars) in `syntax/yacc.vim`:
```
SynFold syn	region	yaccInit2      start='\%^.'ms=s-1,rs=s-1	matchgroup=yaccSectionSep	end='^%%$'me=e-2,re=e-2	contains=@yaccInitCluster	nextgroup=yaccRules	skipwhite skipempty
```
specifically, the `end='^%%$'`

**Comment by chrisbra** [comment]
Please contact the syntax file maintainer.

**Comment by chrisbra** [comment]
seems to be fixed as of 723dd94
